### PR TITLE
K8s mm less fixes

### PR DESCRIPTION
**Pull request by nlippis**
### Update Fabric8 version and allow metrics monitors to be overriden

This PR pull in the same code for https://github.com/apache/druid/pull/13804 and includes the code to upgrade the PodTemplateTaskAdapter.

This PR contains 2 items.

1. Upgrade the k8s client to support newer versions of k8s.
2. Ability to override specific Metrics Monitors
We found that when we used the TaskCountStatsMonitor in the overlord config, the peon tasks would not start because they inherit the monitors from the parent process, which used to be the Middle Manager (but that would never have that monitor originally). So now you can override this value with the following config:

`druid.indexer.runner.peonMonitors`

#### Release note

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesPeonClient.java [hunk 1]
```diff
@@ -106,7 +104,8 @@ public JobResponse waitForJobCompletion(K8sTaskId taskId, long howLong, TimeUnit
                       .inNamespace(namespace)
                       .withName(taskId.getK8sTaskId())
                       .waitUntilCondition(
-                          x -> (x == null) || (x.getStatus() != null && x.getStatus().getActive() == null),
+                          x -> (x == null) || (x.getStatus() != null && x.getStatus().getActive() == null
+                          && (x.getStatus().getFailed() != null || x.getStatus().getSucceeded() != null)),
```
Comment (georgew5656) [inline extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesPeonClient.java:108]: this makes sense to me but what was the reasoning for adding this?

Comment (nlippis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesPeonClient.java:108]: More context [here](https://github.com/apache/druid/pull/13804#discussion_r1112253214)

File: extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java [hunk 2]
```diff
@@ -240,6 +240,7 @@ JobResponse monitorJob(Pod peonPod, K8sTaskId k8sTaskId)
   @Override
   public void updateStatus(Task task, TaskStatus status)
   {
+    log.info("Updating task: %s with status %s", task.getId(), status);
```
[the remark this answers is not in the document]
Comment (nlippis) [reply extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java:243]: This was part of the original PR, IMO this should be a debug log.

**Review by abhishekagarwal87 — APPROVED** [review]
Reviewed the additional commits on top of the previous PR (https://github.com/apache/druid/pull/13804) that was already approved.

**Pull request merged**